dev/com.ibm.ws.security.mp.jwt_fat/fat/src/com/ibm/ws/security/mp/jwt/fat/sharedTests/MPJwtMPConfigTests.java

```diff
@@ -0,0 +1,628 @@
+                                 "com/ibm/ws/security/jwt/fat/mpjwt/CommonMicroProfileApp.class")
+                            .add(new FileAsset(new File("build/classes/com/ibm/ws/security/jwt/fat/mpjwt/MpJwtFatConstants.class")),
+                                 "com/ibm/ws/security/jwt/fat/mpjwt/MpJwtFatConstants.class")
+                            .add(new FileAsset(new File("test-applications/" + warName + "/resources/WEB-INF/web.xml")), "/WEB-INF/web.xml")
+                            .add(new FileAsset(new File("test-applications/" + warName + "/resources/META-INF/permissions.xml")), "/META-INF/permissions.xml")
+                            .add(new FileAsset(new File("publish/shared/securityKeys/bad_key.pem")), "/WEB-INF/classes/bad_key.pem")
+                            .add(new FileAsset(new File("publish/shared/securityKeys/rsa_key_withCert.pem")), "/WEB-INF/classes/rsa_key_withCert.pem")
+                            .add(new FileAsset(new File("publish/shared/securityKeys/rsa_key.pem")), "/WEB-INF/classes/rsa_key.pem");
+            for (String theClass : classList) {
+                newWar.addClass(theClass);
+            }
+            return newWar;
+        } catch (Exception e) {
+            Log.error(thisClass, "genericCreateArchive", e);
+            throw e;
+        }
+    }
+
+    protected static WebArchive genericCreateArchiveWithPemsAndMPConfig(String sourceName, String baseWarName, List<String> classList, String mpConfig,
+                                                                        String fileContent) throws Exception {
+        try {
+            String warName = baseWarName + ".war";
+            WebArchive newWar = ShrinkWrap.create(WebArchive.class, warName)
+                            .add(new FileAsset(new File("build/classes/com/ibm/ws/security/jwt/fat/mpjwt/CommonMicroProfileApp.class")),
+                                 "com/ibm/ws/security/jwt/fat/mpjwt/CommonMicroProfileApp.class")
+                            .add(new FileAsset(new File("build/classes/com/ibm/ws/security/jwt/fat/mpjwt/MpJwtFatConstants.class")),
+                                 "com/ibm/ws/security/jwt/fat/mpjwt/MpJwtFatConstants.class")
+                            .add(new FileAsset(new File("test-applications/" + sourceName + "/resources/WEB-INF/web.xml")), "/WEB-INF/web.xml")
+                            .add(new FileAsset(new File("test-applications/" + sourceName + "/resources/META-INF/permissions.xml")), "/META-INF/permissions.xml")
+                            .add(new FileAsset(new File("publish/shared/securityKeys/bad_key.pem")), "/WEB-INF/classes/bad_key.pem")
+                            .add(new FileAsset(new File("publish/shared/securityKeys/rsa_key_withCert.pem")), "/WEB-INF/classes/rsa_key_withCert.pem")
+                            .add(new FileAsset(new File("publish/shared/securityKeys/rsa_key.pem")), "/WEB-INF/classes/rsa_key.pem")
```

Comment:
Not sure if this class is still in flux, but this is another place where some of these `WebArchive` creation steps can be extracted to a common method.